src/batched/dense/impl/KokkosBatched_Gesv_Impl.hpp

```diff
@@ -0,0 +1,548 @@
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+#ifndef __KOKKOSBATCHED_GESV_IMPL_HPP__
+#define __KOKKOSBATCHED_GESV_IMPL_HPP__
+
+/// \author Kim Liegeois (knliege@sandia.gov)
+
+#include "KokkosBatched_Util.hpp"
+#include <KokkosBatched_LU_Decl.hpp>
+#include "KokkosBatched_Trsm_Decl.hpp"
+
+namespace KokkosBatched {
+
+struct SerialStaticPivoting {
+  template <class MatrixType1, class MatrixType2, class VectorType1,
+            class VectorType2>
+  KOKKOS_INLINE_FUNCTION static void invoke(
+      const MatrixType1 A, const MatrixType2 PDAD, const VectorType1 Y,
+      const VectorType2 PDY, const VectorType2 D2, const VectorType2 tmp_v_1,
+      const VectorType2 tmp_v_2);
+};
+
+template <typename MemberType>
+struct TeamStaticPivoting {
+  template <class MatrixType1, class MatrixType2, class VectorType1,
+            class VectorType2>
+  KOKKOS_INLINE_FUNCTION static void invoke(
+      const MemberType &member, const MatrixType1 A, const MatrixType2 PDAD,
+      const VectorType1 Y, const VectorType2 PDY, const VectorType2 D2,
+      const VectorType2 tmp_v_1, const VectorType2 tmp_v_2);
+};
+
+template <typename MemberType>
+struct TeamVectorStaticPivoting {
+  template <class MatrixType1, class MatrixType2, class VectorType1,
+            class VectorType2>
+  KOKKOS_INLINE_FUNCTION static void invoke(
+      const MemberType &member, const MatrixType1 A, const MatrixType2 PDAD,
+      const VectorType1 Y, const VectorType2 PDY, const VectorType2 D2,
+      const VectorType2 tmp_v_1, const VectorType2 tmp_v_2);
+};
+
+template <class MatrixType1, class MatrixType2, class VectorType1,
+          class VectorType2>
+KOKKOS_INLINE_FUNCTION void SerialStaticPivoting::invoke(
+    const MatrixType1 A, const MatrixType2 PDAD, const VectorType1 Y,
+    const VectorType2 PDY, const VectorType2 D2, const VectorType2 tmp_v_1,
+    const VectorType2 tmp_v_2) {
+  using value_type = typename MatrixType1::non_const_value_type;
+  const int n      = A.extent(0);
```

Comment:
Does `View::extent()` always return `int`, I thought on some architectures it might be `size_t`?